### PR TITLE
Add the to_bytes consistency test from recrypt scala

### PR DESCRIPTION
**Pull request by coltfred**

### CODE REVIEW

File: src/internal/double.rs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+
```
Comment (ernieturner) [inline src/internal/double.rs]: Unnecessary whitespace?

**Comment by coltfred** [comment]
The value is the hashable byte value of the reencrypted value. In rust that'd be `.to_bytes` and in scala it'd be `.to_hash_bytes`. Really this assertion is to make sure we don't break this since it'd break compatibility. I printed it in hex there and pasted it in here.
